ext/date/php_date_arginfo.h

```diff
@@ -700,13 +702,124 @@ static const zend_function_entry class_DatePeriod_methods[] = {
+	zend_string *const_RFC1036_value_str = zend_string_init(DATE_FORMAT_RFC1036, sizeof(DATE_FORMAT_RFC1036) - 1, 1);
+	ZVAL_STR(&const_RFC1036_value, const_RFC1036_value_str);
+	zend_string *const_RFC1036_name = zend_string_init_interned("RFC1036", sizeof("RFC1036") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC1036_name, &const_RFC1036_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC1036_name);
+
+	zval const_RFC1123_value;
+	zend_string *const_RFC1123_value_str = zend_string_init(DATE_FORMAT_RFC1123, sizeof(DATE_FORMAT_RFC1123) - 1, 1);
+	ZVAL_STR(&const_RFC1123_value, const_RFC1123_value_str);
+	zend_string *const_RFC1123_name = zend_string_init_interned("RFC1123", sizeof("RFC1123") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC1123_name, &const_RFC1123_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC1123_name);
+
+	zval const_RFC7231_value;
+	zend_string *const_RFC7231_value_str = zend_string_init(DATE_FORMAT_RFC7231, sizeof(DATE_FORMAT_RFC7231) - 1, 1);
+	ZVAL_STR(&const_RFC7231_value, const_RFC7231_value_str);
+	zend_string *const_RFC7231_name = zend_string_init_interned("RFC7231", sizeof("RFC7231") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC7231_name, &const_RFC7231_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC7231_name);
+
+	zval const_RFC2822_value;
+	zend_string *const_RFC2822_value_str = zend_string_init(DATE_FORMAT_RFC2822, sizeof(DATE_FORMAT_RFC2822) - 1, 1);
+	ZVAL_STR(&const_RFC2822_value, const_RFC2822_value_str);
+	zend_string *const_RFC2822_name = zend_string_init_interned("RFC2822", sizeof("RFC2822") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC2822_name, &const_RFC2822_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC2822_name);
+
+	zval const_RFC3339_value;
+	zend_string *const_RFC3339_value_str = zend_string_init(DATE_FORMAT_RFC3339, sizeof(DATE_FORMAT_RFC3339) - 1, 1);
+	ZVAL_STR(&const_RFC3339_value, const_RFC3339_value_str);
+	zend_string *const_RFC3339_name = zend_string_init_interned("RFC3339", sizeof("RFC3339") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC3339_name, &const_RFC3339_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC3339_name);
+
+	zval const_RFC3339_EXTENDED_value;
+	zend_string *const_RFC3339_EXTENDED_value_str = zend_string_init(DATE_FORMAT_RFC3339_EXTENDED, sizeof(DATE_FORMAT_RFC3339_EXTENDED) - 1, 1);
+	ZVAL_STR(&const_RFC3339_EXTENDED_value, const_RFC3339_EXTENDED_value_str);
+	zend_string *const_RFC3339_EXTENDED_name = zend_string_init_interned("RFC3339_EXTENDED", sizeof("RFC3339_EXTENDED") - 1, 1);
+	zend_declare_class_constant_ex(class_entry, const_RFC3339_EXTENDED_name, &const_RFC3339_EXTENDED_value, ZEND_ACC_PUBLIC, NULL);
+	zend_string_release(const_RFC3339_EXTENDED_name);
+
+	zval const_RSS_value;
+	zend_string *const_RSS_value_str = zend_string_init("D, d M Y H:i:s O", sizeof("D, d M Y H:i:s O") - 1, 1);
```

Comment:
Nice catch! I'll have to rework the value evaluator to fix this issue in a maintainable way.

Reply:
So the issue is that currently, C constants are only referenced in the generated code when they have a C name (`@cname`). Since `DATE_RSS` is just an alias for `DATE_RFC1123`, it doesn't need an underlying C constant, so it doesn't have a name -> that's why the literal value is generated.

This issue could be easily fixed by following the "originating" constant chain, and reference the first one which has a name. My problem with this approach is what you also noticed in one of your review comments: it would fall short when multiple constants are referenced in an expression (i.e. `E_ALL`).

On one hand, we can decide so that we don't support expressions (at least for the time being), and go with the above approach, but I'm not sure if it's a good idea to add more code which most probably needs to be changed later. On the other hand, a quick-fix would be to simply generate an underlying C constant for these values, at problem is solved.

However, since this is really just generated code, I am not sure if it's crucial to reference exactly the same values as what's in the stubs (or their C constant counterparts). Do you have any stong(ish) preference?